README.md

```diff
@@ -1,180 +1,220 @@
-~~~~~~{.sh}
-
-# get ubuntu packages
-# clang-3.6 can be replaced by any later versions of clang
-sudo apt-get install clang-3.6 doxygen build-essential thrift-compiler python-virtualenv gcc-multilib git wget python-dev
-
-# create new python virtualenv
+[![Build status](https://travis-ci.org/Ericsson/codechecker.svg?branch=master)](https://travis-ci.org/Ericsson/codechecker)
+
+CodeChecker
+===========
+
+**CodeChecker** is a static analysis infrastructure built on the [LLVM/Clang
+Static Analyzer](http://clang-analyzer.llvm.org) toolchain, replacing
+[`scan-build`](http://clang-analyzer.llvm.org/scan-build.html) in a Linux
+development environment.
+
+
+![CodeChecker web interface for viewing discovered code defects](https://raw.githubusercontent.com/Ericsson/codechecker/master/docs/images/viewer.png)
+
+
+Main features
+-------------
+
+  * Support for multiple analyzers, currently _Clang Static Analyzer_ and
+    _Clang-Tidy_
+  * Store results of multiple large-scale analysis runs efficiently, either in
+    a PostgreSQL or SQLite database
+  * **Web application** for viewing discovered code defects with a streamlined,
+    easy experience
+  * **Filterable** (defect checker name, severity, source paths, ...) and
+    **comparable** (calculates difference between two analyses of the project,
+    showing which bugs have been fixed and which are newly introduced) result
+    viewing
+  * Subsequent analysis runs **only check** and update results for **modified
+    files** without analysing the _entire_ project (depends on build toolchain
+    support!)
+  * Suppression of known false positive results, either in configuration file
+    or via annotation in source code, along with exclusion of entire source
+    paths from analysis
+  * `quickcheck` mode shows analysis results on standard output
+  * Easily implementable [Thrift](http://thrift.apache.org)-based
+    server-client communication used for storing and querying of discovered
+    defects
+  * Support for multiple bug visualisation frontends, such as the web
+    application, a [command-line tool](docs/usage.md) and an [Eclipse
+    plugin](http://github.com/Ericsson/CodeCheckerEclipsePlugin)
+
+
+A high-level overview about the infrastructure is available amongst the [2015
+Euro LLVM Conference](http://llvm.org/devmtg/2015-04) presentations.<br/>
+**Dániel KRUPP, György ORBÁN, Gábor HORVÁTH and Bence BABATI**:<br/>
+[_Industrial Experiences with the Clang Static Analysis Toolset_](http://llvm.org/devmtg/2015-04/slides/Clang_static_analysis_toolset_final.pdf)
+
+Install
+-------
+
+### Linux
+
+For a detailed dependency list, please see [Requirements](docs/deps.md). The
+following commands are used to bootstrap CodeChecker on Ubuntu 14.04.2 LTS:
+
+```bash
+# Install mandatory dependencies for a development and analysis environment
+# NOTE: clang-3.6 can be replaced by any later versions of LLVM/Clang
+sudo apt-get install clang-3.6 build-essential doxygen gcc-multilib git \
```

Comment:
I think since clang 3.8 is available in xenial (16.04LTS) we can bump up the required clang version here to clang-3.8 and clang-tidy-3.8